ReSharper.FSharp/src/FSharp.Psi.Features/src/CodeCleanup/FantomasFormatterProvider.fs

```diff
@@ -3,15 +3,31 @@ namespace JetBrains.ReSharper.Plugins.FSharp.Services.Formatter
 open System
 open FSharp.Compiler.CodeAnalysis
 open FSharp.Compiler.Text
+open JetBrains.Core
 open JetBrains.Lifetimes
 open JetBrains.ProjectModel
 open JetBrains.Rd.Tasks
 open JetBrains.ReSharper.Plugins.FSharp.Fantomas.Client
 open JetBrains.ReSharper.Plugins.FSharp.Fantomas.Protocol
 
+module internal Reflection =
+    let formatSettingType = typeof<FSharpFormatSettingsKey>
+
+    let getFieldValue obj fieldName =
+        let field = formatSettingType.GetField(fieldName)
+        if isNotNull field then field.GetValue(obj) else null
+
+
 [<SolutionComponent>]
 type FantomasFormatterProvider(solution: ISolution, fantomasFactory: FantomasProcessFactory) =
     let mutable connection: FantomasConnection = null
+    let mutable formatConfigFields: string[] = [||]
+
+    let toEditorConfigName value =
+        value
+        |> Seq.map (fun c -> if Char.IsUpper(c) then $"_%s{c.ToString().ToLower()}" else c.ToString())
+        |> String.concat ""
+        |> sprintf "fsharp%s"
```

Comment:
Could you add a test for these conversions?

Reply:
Also, please add some comments, because it's not clear what is being done here.

Reply:
BTW, we have StringUtil.MakeUnderscoreCaseName, if you wanted to do that